Sources/NIOFileSystem/FileChunks.swift

```diff
@@ -85,97 +85,168 @@ extension FileChunks.FileChunkIterator: Sendable {}
 // MARK: - Internal
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-extension BufferedStream where Element == ByteBuffer {
+private typealias FileChunkSequenceProducer = NIOThrowingAsyncSequenceProducer<
+    ByteBuffer, Error, NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark, FileChunkProducer
+>
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension NIOThrowingAsyncSequenceProducer
+where
+    Element == ByteBuffer,
+    Failure == Error,
+    Strategy == NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark,
+    Delegate == FileChunkProducer
+{
     static func makeFileChunksStream(
         of: Element.Type = Element.self,
         handle: SystemFileHandle,
         chunkLength: Int64,
         range: FileChunks.ChunkRange,
         lowWatermark: Int,
         highWatermark: Int
-    ) -> BufferedStream<ByteBuffer> {
-        let state: ProducerState
-        switch range {
-        case .entireFile:
-            state = ProducerState(handle: handle, range: nil)
-        case .partial(let partialRange):
-            state = ProducerState(handle: handle, range: partialRange)
-        }
-        let protectedState = NIOLockedValueBox(state)
+    ) -> FileChunkSequenceProducer {
 
-        var (stream, source) = BufferedStream.makeStream(
-            of: ByteBuffer.self,
-            backPressureStrategy: .watermark(low: lowWatermark, high: highWatermark)
+        let producer = FileChunkProducer(
+            range: range,
+            handle: handle,
+            chunkLength: chunkLength
         )
 
-        source.onTermination = {
-            protectedState.withLockedValue { state in
-                state.done()
-            }
-        }
-
-        // Start producing immediately.
-        let producer = FileChunkProducer(
-            state: protectedState,
-            source: source,
-            path: handle.path,
-            length: chunkLength
+        let nioThrowingAsyncSequence = NIOThrowingAsyncSequenceProducer.makeSequence(
+            elementType: ByteBuffer.self,
+            backPressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark(
+                lowWatermark: 4,
+                highWatermark: 8
```

Comment:
We pass in the watermarks to `makeFileChunksStream` so presumably we should use them here?

Reply:
Good catch!